packages/fxa-auth-server/test/local/payments/stripe.js

```diff
@@ -1322,8 +1302,94 @@ describe('StripeHelper', () => {
     });
   });
 
+  describe('fetchProductById', () => {
+    let mockStripe;
+    const planId = 'plan_00000000000000';
+    const planName = 'Example Plan';
+    const productId = 'prod_00000000000000';
+    const productName = 'Example Product';
+    const planEmailIconURL = 'http://example.com/icon';
+    const planDownloadURL = 'http://example.com/download';
+    const privacyNoticeURL =
+      'https://www.mozilla.org/privacy/firefox-private-network';
+    const termsOfServiceURL =
+      'https://www.mozilla.org/about/legal/terms/firefox-private-network';
+
+    const mockPlan = {
+      id: planId,
+      nickname: planName,
+      product: productId,
+      metadata: {
+        emailIconURL: planEmailIconURL,
+        downloadURL: planDownloadURL,
+      },
+    };
+
+    const mockProduct = {
+      id: productId,
+      name: productName,
+      metadata: {
+        'product:termsOfServiceURL': termsOfServiceURL,
+        'product:privacyNoticeURL': privacyNoticeURL,
+      },
+    };
+    beforeEach(() => {
+      mockStripe = Object.entries({
+        plans: mockPlan,
+        products: mockProduct,
+      }).reduce(
+        (acc, [resource, value]) => ({
+          ...acc,
+          [resource]: { retrieve: sinon.stub().resolves(value) },
+        }),
+        {}
+      );
+      stripeHelper.stripe = mockStripe;
+    });
+
+    it('returns null if there is an issue fetching the product', async () => {
+      stripeHelper.stripe.products.retrieve.returns(
+        Promise.reject(new Error('product does not exist'))
+      );
```

Comment:
I think this can use `stripeHelper.stripe.products.retrieve.rejects`.  Just fyi; this works too.